include/crm/common/util.h

```diff
@@ -166,7 +165,7 @@ int crm_user_lookup(const char *name, uid_t * uid, gid_t * gid);
 void crm_gnutls_global_init(void);
 #endif
 
-int crm_exit(int rc);
+crm_exit_t crm_exit(crm_exit_t rc);
```

Comment:
sbd/src/sbd-inquisitor.c includes crm/common/util.h but now is encountering:
```
/usr/include/pacemaker/crm/common/util.h:168:1: error: unknown type name 'crm_exit_t'
```
The type is defined in crm/crm.h which includes this header though... I don't know what is the best solution ;)

BTW crm_internal.h  references this type too.

Reply:
Recursive includes wouldn't really be a problem thanks to the #ifdef guards.

But it would be nice for 2.0 to organize the headers a lot better.

I'm torn between fewer or more headers. Fewer = two headers (one public and one private) for each library, e.g. include/crm/common.h and include/crm/common_internal.h would declare everything in libcrmcommon. More = divide headers more finely by functionality, e.g. for this issue, create an include/crm/common/exit.h and include it in both util.h and crm.h. (Moving crm_exit_t to error.h, and including that in util.h, would be another option.)

Reply:
sbd should build against 2.0 again ... crm_exit_t, related functions, and the old error.h are now in include/crm/common/results.h which is now included in util.h.

Reply:
Yes, sbd compiles again. Thanks.